iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -260,7 +260,7 @@ template <typename T>
 inline T& optional<T>::value() & noexcept
 {
     Expects(has_value());
-    return *static_cast<T*>(static_cast<void*>(m_data));
+    return *static_cast<T*>(static_cast<void*>(&m_data));
```

Comment:
I hope this helps with QNX @dkroenke as well. Otherwise `reinterpret_cast` might help, but there is no unaligned access with this change at least.

`reinterpret_cast` would require a clang_tidy comment though.